test/parallel/test-http-client-set-timeout-after-end.js

```diff
@@ -0,0 +1,69 @@
+'use strict';
```

Comment:
Can I suggest a simpler test? I think this one is overcomplicated.

```js
'use strict';

const { mustCall } = require('../common');

const { Agent, createServer, get } = require('http');
const { strictEqual} = require('assert');

const server = createServer(mustCall((req, res) => {
  res.end();
}));

server.listen(0, () => {
  const agent = new Agent({ keepAlive: true, maxSockets: 1 });
  const port = server.address().port;

  let socket;

  const req = get({ agent, port }, (res) => {
    res.on('end', () => {
      strictEqual(req.setTimeout(0), req);
      strictEqual(socket.listenerCount('timeout'), 0);
      agent.destroy();
      server.close();
    });
    res.resume();
  });

  req.on('socket', (sock) => {
    socket = sock;
  });
});
```

Reply:
It's based on @szmarczak's reproduction of the issue (plus an existing test) so I wanted to stick to that to avoid the risk of reproducing a different issue. Suggestions?

Reply:
IMO @lpinca's test is good. It reflects the test I [wrote for Got](https://github.com/sindresorhus/got/commit/203dadcfc41753aa515068032cd1f30f6945fae1#diff-f7436169f25ce731ecf3d9892df46eb4R477).

Reply:
Fair enough. I'll paste it in tomorrow as I'm on my phone right now. :-)

Reply:
Pasted and force-pushed as 131514be0a.

Reply:
Thank you.